backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/window/descriptor/AdvancedSearchBPartnerProcessor.java

```diff
@@ -64,6 +67,13 @@ public void processSelection(
 		}
 		document.processValueChange(bpartnerFieldName, bpartnerId, null, false);
 
+		final DocumentEntityDescriptor entityDescriptor = document.getEntityDescriptor();
+		final String tableName = entityDescriptor.getTableName();
+		final int adTableId = adTableDAO.retrieveTableId(tableName);
+		final int bPartnerTableId = adTableDAO.retrieveTableId(I_C_BPartner.Table_Name);
+		if ((adTableId == bPartnerTableId) && (bpartnerFieldName.equals(I_C_BPartner.COLUMNNAME_C_BPartner_SalesRep_ID)))
```

Comment:
1. Please document what's the logic behind this? for me it's not obvious
2. why simply comparing by tableName is not enough? Why you have to get the ad_table_id and compare by that?

Reply:
1. The logic is to ignore the location and contact update, when we selection a `SalesRep` in a `BPartner` window
2. Ok, just comparing by Table Name